src/event.js

```diff
@@ -392,90 +422,89 @@ jQuery.event = {
+			}
 
-	mouseHooks: {
-		props: ( "button buttons clientX clientY offsetX offsetY pageX pageY " +
-			"screenX screenY toElement" ).split( " " ),
-		filter: function( event, original ) {
-			var eventDoc, doc, body,
-				button = original.button;
+			return event.which;
+		},
 
-			// Calculate pageX/Y if missing and clientX/Y available
-			if ( event.pageX == null && original.clientX != null ) {
+		pageX: function( event ) {
+			var eventDoc, doc, body;
+
+			// Calculate pageX if missing and clientX available
+			if ( event.pageX == null && event.clientX != null ) {
 				eventDoc = event.target.ownerDocument || document;
 				doc = eventDoc.documentElement;
 				body = eventDoc.body;
 
-				event.pageX = original.clientX +
+				return event.clientX +
 					( doc && doc.scrollLeft || body && body.scrollLeft || 0 ) -
 					( doc && doc.clientLeft || body && body.clientLeft || 0 );
-				event.pageY = original.clientY +
-					( doc && doc.scrollTop  || body && body.scrollTop  || 0 ) -
-					( doc && doc.clientTop  || body && body.clientTop  || 0 );
 			}
 
-			// Add which for click: 1 === left; 2 === middle; 3 === right
-			// Note: button is not normalized, so don't use it
-			if ( !event.which && button !== undefined ) {
-				event.which = ( button & 1 ? 1 : ( button & 2 ? 3 : ( button & 4 ? 2 : 0 ) ) );
+			return event.pageX;
+		},
+
+		pageY: function( event ) {
+			var eventDoc, doc, body;
+
+			// Calculate pageY if missing and clientY available
+			if ( event.pageY == null && event.clientY != null ) {
+				eventDoc = event.target.ownerDocument || document;
+				doc = eventDoc.documentElement;
+				body = eventDoc.body;
+
+				return event.clientY +
+					( doc && doc.scrollTop || body && body.scrollTop || 0 ) -
+					( doc && doc.clientTop || body && body.clientTop || 0 );
 			}
 
-			return event;
+			return event.pageY;
 		}
 	},
 
-	fix: function( event ) {
-		if ( event[ jQuery.expando ] ) {
-			return event;
+	fix: function( originalEvent ) {
+		if ( originalEvent[ jQuery.expando ] ) {
+			return originalEvent;
 		}
 
 		// Create a writable copy of the event object and normalize some properties
-		var i, prop, copy,
-			type = event.type,
-			originalEvent = event,
-			fixHook = this.fixHooks[ type ];
-
-		if ( !fixHook ) {
-			this.fixHooks[ type ] = fixHook =
-				rmouseEvent.test( type ) ? this.mouseHooks :
-				rkeyEvent.test( type ) ? this.keyHooks :
-				{};
+		var event,
+			fixHook = this.fixHooks[ originalEvent.type ];
+
+		// Setup any prop hooks added since the last fix
+		if ( this.props.length ) {
+			jQuery.each( this.props.splice( 0 ), addEventGetter );
 		}
-		copy = fixHook.props ? this.props.concat( fixHook.props ) : this.props;
 
 		event = new jQuery.Event( originalEvent );
 
-		i = copy.length;
-		while ( i-- ) {
-			prop = copy[ i ];
-			event[ prop ] = originalEvent[ prop ];
-		}
-
 		// Support: Safari 6-8+
 		// Target should not be a text node (#504, #13143)
 		if ( event.target.nodeType === 3 ) {
```

Comment:
It would be the caller's error for them to pass something other than an element to `jQuery.Event` though, and they control if that happens. For this case we're working around a browser-generated event so although we can blame Safari we'll never get them to commit to fixing it. :smile_cat: